src/GridFS/StreamWrapper.php

```diff
@@ -278,26 +329,24 @@ private function getStatTemplate(): array
      *
      * @see StreamWrapper::stream_open()
      */
-    private function initProtocol(string $path): void
+    private function initProtocol(string $path): string
```

Comment:
If this is no longer setting a `$protocol` property, I think we should either rename this to `getProtocol()` or simply inline it within `stream_open()`. The latter probably makes sense if this is only called once.

This could just be a one-liner:

```
$protocol = explode('://', $path, 2)[0] ?: 'gridfs';
```

That said, while [testing this](https://3v4l.org/TIMYf#v7.4.33) it doesn't look like it does much. If a completely bogus path is provided without a scheme, the path itself is going to be picked up as the protocol.

It's been a while since I worked with the stream wrapper, so perhaps this isn't anything to worry about since we'd only ever receive a path _with_ a scheme. But in that case, I'm nto sure why we ever worry about using `?:` here. `explode()` only returns an empty array for empty input. Any string without a schema is going to return a single-element array with that string.